src/middleware/jira-admin-permission-middleware.ts

```diff
@@ -0,0 +1,48 @@
+import { NextFunction, Request, Response } from "express";
+import { Installation } from "models/installation";
+import { JiraClient } from "models/jira-client";
+import { booleanFlag, BooleanFlags } from "config/feature-flags";
+
+export const setJiraAdminPrivileges = async (req: Request, claims: Record<any, any>, installation: Installation) => {
+	const ADMIN_PERMISSION = "ADMINISTER";
+	// We only need to add this to the session if it doesn't exist
+	if (req.session.isJiraAdmin !== undefined) {
+		return;
+	}
+
+	try {
+		const userAccountId = claims.sub;
+		// Can't check permissions without userAccountId
+		if (!userAccountId) {
+			return;
+		}
+		const jiraClient = await JiraClient.getNewClient(installation, req.log);
+		// Make jira call to permissions with userAccountId.
+		const permissions = await jiraClient.checkAdminPermissions(userAccountId);
+		const hasAdminPermissions = permissions.data.globalPermissions.includes(ADMIN_PERMISSION);
+
+		req.session.isJiraAdmin = hasAdminPermissions;
+		req.log.info({ isAdmin :req.session.isJiraAdmin }, "Admin permissions set");
+	} catch (err) {
+		req.log.error({ err }, "Failed to fetch Jira Admin rights");
+	}
+};
+
+export const jiraAdminPermissionsMiddleware = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Why do we rely on the session value hasAdminPermissions instead of reading it directyl from the token??

I agree, mostly, this means we a tightly coupled with the JWT middleware and that we need to put this middleware after every time, kinda annoying, but have put in some comments and messaging to help raise that if it happens.

The problem being, if we decide to decouple and decode the token with cert here, its duplicating the code of the JWT middleware so then we would need to refactor that so its reuseable in both places, while I don't disagree with that approach, the effort was too high and this was imo the best trade off. 
Open to opinions tho!